### ISSUE TITLE
False positive from clippy when adding `#[allow(clippy::needless_lifetimes)]` annotation.

### ISSUE THREAD

**Issue by Santoi** [rust-lang/rust-clippy#14441]
### Summary

For reference: https://github.com/maliput/maliput-rs/pull/142

I want to evade what I already think is a false positive with an `#[allow(...)]` annotation.

I have 2 structs that are the same, which has never been an issue so far, but a solution for a problem I'm not familiar with. They are under different namespaces, but I feel like it may have something to do.

I added the annotation before both of them, and there are no other annotations.

The structs are in a mod that is annotated with `#[cxx::bridge(namespace = "maliput::api")]`

### Lint Name

needless_lifetimes

### Reproducer

I tried this code:

```rust
#[allow(clippy::needless_lifetimes)]
struct ConstLaneSRangeRef<'a> {
    pub lane_s_range: &'a LaneSRange,
}
```

I saw this happen: (yes, same error twice in a row)

```
error: the following explicit lifetimes could be elided: 'a
  --> maliput-sys/src/api/mod.rs:48:31
   |
48 |     struct ConstLaneSRangeRef<'a> {
   |                               ^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
help: elide the lifetimes
   |
48 -     struct ConstLaneSRangeRef<'a> {
48 +     struct ConstLaneSRangeRef'_> {
   |

error: the following explicit lifetimes could be elided: 'a
  --> maliput-sys/src/api/mod.rs:48:31
   |
48 |     struct ConstLaneSRangeRef<'a> {
   |                               ^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
help: elide the lifetimes
   |
48 -     struct ConstLaneSRangeRef<'a> {
48 +     struct ConstLaneSRangeRef'_> {
   |

```

I expected to see this happen:

🟢 Green lights.

### Version

```text
rustc 1.83.0 (90b35a623 2024-11-26)
binary: rustc
commit-hash: 90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf
commit-date: 2024-11-26
host: x86_64-unknown-linux-gnu
release: 1.83.0
LLVM version: 19.1.1
```
